### PR TITLE
Fixing blob size header validation issue

### PR DESCRIPTION
**Pull request by cgtz**
Previously, providing blob size headers with non-parsable values would result in empty responses. This fixes the issue so that the response code is correctly set to 400.

*Time to review:* 5 min.
*Primary reviewers:* @nsivabalan, @vgkholla 
*Testing:* Unit tests and local deployment/curl testing

### CODE REVIEW

File: ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java [hunk 1]
```diff
@@ -703,4 +704,19 @@ public void getTimeFromDateStringTest() {
 
     assertEquals("Should have returned null", null, RestUtils.getTimeFromDateString("abc"));
   }
+
+  @Test
```
Comment (vgkholla) [inline ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java]: doc

File: ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java [hunk 2]
```diff
@@ -448,6 +435,27 @@ public static Long getTimeFromDateString(String dateString) {
   }
 
   /**
+   * Parse a blob size string and return the blob size as a number, if valid.
+   * @param blobSizeStr a string representing the blob size.
+   * @return the blob size as a {@code long}.
+   * @throws RestServiceException if a valid blob size could not be parsed.
+   */
+  public static long getBlobSize(String blobSizeStr)
```
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java]: why is this `public`?

Comment (cgtz) [inline ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java]: I can make this package-private

File: ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java (hunk 1 above)
Comment (cgtz) [inline ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java]: added

File: ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java [hunk 3]
```diff
@@ -29,6 +29,7 @@
 import java.util.Map;
 import java.util.Random;
 import java.util.TimeZone;
+import junit.framework.Assert;
```
Comment (vgkholla) [inline ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java]: why was this added if no new code was added? Is this used?

**Comment by vgkholla** [comment]
@cgtz Please check if you have formatted all the files and please confirm that build passes.

@nsivabalan Please merge if this looks good to you.

File: ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java [hunk 4]
```diff
@@ -137,8 +137,19 @@ public NettyRequest(HttpRequest request, Channel channel, NettyMetrics nettyMetr
...
     }
 
-    if (HttpHeaders.getHeader(request, RestUtils.Headers.BLOB_SIZE, null) != null) {
-      size = Long.parseLong(HttpHeaders.getHeader(request, RestUtils.Headers.BLOB_SIZE));
+    String blobSizeStr = HttpHeaders.getHeader(request, RestUtils.Headers.BLOB_SIZE, null);
+    if (blobSizeStr != null) {
+      try {
+        size = Long.parseLong(blobSizeStr);
+        if (size < 0) {
+          throw new RestServiceException(RestUtils.Headers.BLOB_SIZE + "[" + size + "] is less than 0",
```
Comment (xiahome) [inline ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java]: Should there be a space between `BLOB_SIZE` and `[`?

File: ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java [hunk 5]
```diff
@@ -137,8 +137,19 @@ public NettyRequest(HttpRequest request, Channel channel, NettyMetrics nettyMetr
...
+              RestServiceErrorCode.InvalidArgs);
+        }
+      } catch (NumberFormatException e) {
+        throw new RestServiceException(
+            RestUtils.Headers.BLOB_SIZE + "[" + blobSizeStr + "] could not parsed into a number",
```
Comment (xiahome) [inline ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java]: Same comment as above.

File: ambry-rest/src/test/java/com.github.ambry.rest/NettyRequestTest.java [hunk 6]
```diff
@@ -206,6 +206,18 @@ public void conversionWithBadInputTest()
     } catch (RestServiceException e) {
       assertEquals("Unexpected RestServiceErrorCode", RestServiceErrorCode.UnsupportedHttpMethod, e.getErrorCode());
     }
+
+    String[] invalidBlobSizeStrs = {"aba123", "12ab", "-1", "ddsdd", "999999999999999999999999999"};
```
Comment (xiahome) [inline ambry-rest/src/test/java/com.github.ambry.rest/NettyRequestTest.java]: Test a decimal number?

File: ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java (hunk 4 above)
Comment (cgtz) [inline ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java]: I'll add a space

**Comment by cgtz** [comment]
@xiahome addressed comments

**Comment by xiahome** [comment]
LGTM, merging.
